src/components/ArticleList.jsx

```diff
@@ -2,20 +2,21 @@ import { useEffect, useState } from "react";
 import fetchArticles from "../controllers/fetchArticles";
 import ArticleListCard from "./ArticleListCard";
 import { Stack } from "@mui/material";
+import { useParams } from "react-router-dom";
 
 function ArticleList() {
+  const { topicName } = useParams();
   const [isLoading, setIsLoading] = useState(true);
-  /** @type {useState<ArticleListState>} */
   const [articles, setArticles] = useState([]);
   useEffect(() => {
     const getArticles = async () => {
       setIsLoading(true);
-      const newArticles = await fetchArticles();
+      const newArticles = await fetchArticles(topicName);
```

Comment:
Love to see you using async/await so effectively in this project - its really common in industry code, so irs great to demonstrate this skill.